hasUniqueChars.c

```diff
@@ -87,29 +73,58 @@ bool hasUniqueChars(char * inputStr) {
   seeBits(checkBitsexcl_amp, debug_str_excl_amp);
   // -------------------------------------------------------------
 
-  // TODO: Declare additional variables you need here
-
-  
   for(i = 0; i < strlen(inputStr); i++) {
     nextChar = inputStr[i];
-    // TODO: Add your code here to check nextChar, see if it is a duplicate, and update the checkBits variables
+    unsigned long index;
+    unsigned long mask;
 
+    // Skip space character
+    if (nextChar == 32) {
+      continue;
+    }
+    // Check if nextChar is between A-Z
+    else if ((nextChar >= 65) && (nextChar <= 90)) {
+      index = nextChar - 65;
+      mask = 1l << index;
+      if (mask & checkBitsA_z) {
+        return false;  // duplicate found
+      } else {
+        checkBitsA_z |= mask; // mark this character as seen
+      }
+    }
+    // Check if nextChar is between a-z
+    else if ((nextChar >= 97) && (nextChar <= 122)) {
+      index = nextChar - 97 + 26;
```

Comment:
Include the characters between 91 and 96, you just need one case for 65 to 126, inclusive